### PR TITLE
[NEW] Reply in direct

### PR DESCRIPTION
**Pull request by GleidsonDaniel**
## Proposed changes

Adds the option to reply to a message sent in a group or team via direct

## How to test or reproduce

- In a group or team, select a message and click "Reply in Direct Message".

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Jira: [TC-121]

[TC-121]: https://rocketchat.atlassian.net/browse/TC-121?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by reinaldonetof — APPROVED** [review]

File: app/containers/MessageActions/index.tsx [hunk 1]
```diff
@@ -237,6 +239,23 @@ const MessageActions = React.memo(
 				replyInit(message, false);
 			};
 
+			const handleReplyInDM = async (message: TAnyMessageModel) => {
+				if (message.u.username) {
```
Comment (diegolmello) [inline app/containers/MessageActions/index.tsx:243]: We should be a little bit more careful about this here.
I'm not sure what could happen if there's no `username` if the message is from some sort of bot or idk.

File: app/containers/MessageActions/index.tsx [hunk 2]
```diff
@@ -237,6 +239,23 @@ const MessageActions = React.memo(
...
+					const result = await Services.createDirectMessage(message.u.username);
+					if (result.success) {
```
Comment (diegolmello) [inline app/containers/MessageActions/index.tsx:245]: Did we test what happens if the DM already exists?

File: app/containers/MessageActions/index.tsx [hunk 3]
```diff
@@ -237,6 +239,23 @@ const MessageActions = React.memo(
...
+						const { room } = result;
+						const params = {
+							rid: room.rid,
+							name: getRoomTitle(room),
+							t: room.t,
+							roomUserId: getUidDirectMessage(room),
+							replyInDM: message
+						};
+						Navigation.replace('RoomView', params);
```
Comment (diegolmello) [inline app/containers/MessageActions/index.tsx:254]: We should try to use `goRoom`

File: app/containers/MessageActions/index.tsx (hunk 2 above)
Comment (reinaldonetof) [reply app/containers/MessageActions/index.tsx:245]: Yeap! Works properly, the user is redirected to the correct DM.

File: app/containers/MessageActions/index.tsx (hunk 3 above)
Comment (reinaldonetof) [reply app/containers/MessageActions/index.tsx:254]: The `goRoom` uses the `Navigation.navigate`, this navigation's type doesn't work when it goes to the "same" view. Also, this handleInDM introduces that behavior that we commented on some time ago, about the navigation between chats.

File: app/containers/MessageActions/index.tsx (hunk 1 above)
Comment (reinaldonetof) [reply app/containers/MessageActions/index.tsx]: Sure, I added the `optional chain operator`

File: app/containers/MessageActions/index.tsx (hunk 3 above)
Comment (diegolmello) [reply app/containers/MessageActions/index.tsx:254]: `goRoom` accepts a `navigationMethod` param.
However remember we're going to deprecate nested rooms soon and we'll have to analyze this case again to use `goRoom` anyway.

**Pull request merged**